src/Microsoft.Health.SqlServer/Features/Client/RetrySqlCommandWrapper.cs

```diff
@@ -15,7 +15,7 @@ namespace Microsoft.Health.SqlServer.Features.Client
     /// <summary>
     /// A wrapper around <see cref="SqlCommand"/> to provide automatic retries for transient errors.
     /// </summary>
-    internal class RetrySqlCommandWrapper : SqlCommandWrapper
+    public class RetrySqlCommandWrapper : SqlCommandWrapper
```

Comment:
Should we also expose `RetrySqlCommandWrapperFactory`?

Reply:
Good question. In the fhir-server, we currently have:

```
services.Add<RetrySqlCommandWrapperFactory>()
                .Singleton()
                .AsSelf()
                .AsService<SqlCommandWrapperFactory>();
```

So we are already using retry SQL logic under the hood. The reason why we are exposing the `RetrySqlCommandWrapper` here is because, in some cases, it is useful to be able to create a custom retry policy to handle certain exceptions that are not marked as transient:

`var retrySqlCommandWrapper = new RetrySqlCommandWrapper(previousSqlCommandWrapper, customRetryPolicy);`

All this said there could definitely be a better way to go about this. I think it might make sense to merge this PR, put up the corresponding fhir-server PR to give a bit more context, and iterate (in either repo) if needed!

Reply:
[Here](https://github.com/microsoft/fhir-server/pull/1811 )is the link to the corresponding fhir-server PR